### PR TITLE
 install UserLogger and use in models Account and Transaction

### PR DESCRIPTION
**Pull request by HamidNiakan**
Install package UserLogger for saving user activites such as:
create account
update account
delete account
create transaction
update transaction
rollback transaction

### CODE REVIEW

File: src/Http/Controllers/AccountController.php [hunk 1]
```diff
@@ -69,6 +71,12 @@ public function store(AccountStoreRequest $request)
             $data['can_receive'],
             $data['meta']
         );
+        $changes = $account->changesForLog();
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($account)
+            ->withProperties($changes)
+            ->log('create-account');
```
Comment (yeganemehr) [inline src/Http/Controllers/AccountController.php]:
```suggestion
            ->log('create');
```

File: src/Http/Controllers/AccountController.php [hunk 2]
```diff
@@ -80,14 +88,27 @@ public function update(Account $account, AccountUpdateRequest $request)
...
         }
-        $this->accountManager->update($account->id, $changes);
+
+        $account = $this->accountManager->update($account->id, $changes);
+        $changes = $account->changesForLog();
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($account)
+            ->withProperties($changes)
+            ->log('update-account');
```
Comment (yeganemehr) [inline src/Http/Controllers/AccountController.php]:
```suggestion
            ->log('update');
```

File: src/Http/Controllers/AccountController.php [hunk 3]
```diff
@@ -80,14 +88,27 @@ public function update(Account $account, AccountUpdateRequest $request)
...
-    public function destroy(Account $account)
+    public function destroy(Account $account, Request $request)
     {
+        $changes = $account->toArray();
         $this->accountManager->delete($account->id);
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($account)
+            ->withProperties($changes)
+            ->log('delete-account');
```
Comment (yeganemehr) [inline src/Http/Controllers/AccountController.php]:
```suggestion
            ->log('delete');
```

File: src/Http/Controllers/TransactionController.php [hunk 4]
```diff
@@ -52,6 +54,12 @@ public function store(TransactionStoreRequest $request)
             $data['meta'] ?? null,
             $data['force'] ?? false,
         );
+        $changes = $transaction->changesForLog();
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($transaction)
+            ->withProperties($changes)
+            ->log('create-transaction');
```
Comment (yeganemehr) [inline src/Http/Controllers/TransactionController.php]:
```suggestion
            ->log('create');
```

File: src/Http/Controllers/TransactionController.php [hunk 5]
```diff
@@ -60,13 +68,25 @@ public function update(Transaction $transaction, TransactionUpdateRequest $reque
     {
         $data = $request->validated();
         $transaction = $this->transactionManager->update($transaction->id, $data['meta']);
+        $changes = $transaction->changesForLog();
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($transaction)
+            ->withProperties($changes)
+            ->log('update-transaction');
```
Comment (yeganemehr) [inline src/Http/Controllers/TransactionController.php]:
```suggestion
            ->log('update');
```

File: src/Http/Controllers/TransactionController.php [hunk 6]
```diff
@@ -60,13 +68,25 @@ public function update(Transaction $transaction, TransactionUpdateRequest $reque
...
-    public function destroy(Transaction $transaction)
+    public function destroy(Transaction $transaction, Request $request)
     {
         $rollback = $this->transactionManager->rollback($transaction->id);
+        $changes = $rollback->changesForLog();
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($transaction)
+            ->withProperties($changes)
+            ->log('rollback-transaction');
```
Comment (yeganemehr) [inline src/Http/Controllers/TransactionController.php]:
```suggestion
            ->log('rollback');
```

**Review by yeganemehr — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/TransactionController.php (hunk 6 above)
Comment (HamidNiakan) [reply src/Http/Controllers/TransactionController.php]: done

File: src/Http/Controllers/TransactionController.php (hunk 5 above)
Comment (HamidNiakan) [reply src/Http/Controllers/TransactionController.php]: done

File: src/Http/Controllers/TransactionController.php (hunk 4 above)
Comment (HamidNiakan) [reply src/Http/Controllers/TransactionController.php]: done

File: src/Http/Controllers/AccountController.php (hunk 3 above)
Comment (HamidNiakan) [reply src/Http/Controllers/AccountController.php]: done

File: src/Http/Controllers/AccountController.php (hunk 2 above)
Comment (HamidNiakan) [reply src/Http/Controllers/AccountController.php]: done

File: src/Http/Controllers/AccountController.php (hunk 1 above)
Comment (HamidNiakan) [reply src/Http/Controllers/AccountController.php]: done

**Review by yeganemehr — APPROVED** [review]

**Pull request merged**